### PR TITLE
Some temporary battle fixes

### PR DESCRIPTION
**Pull request by Ultima1990**
Adds Ultra Wormhole sprite. Also used for Ultra Megalopolis. Very ugly. Anyone is free to make something better.
Turned Ultra Megalopolis into a one time battle. Removed Elite_UltraNecrozma badge.
Changed Eater of Light quest to target the TemporaryBattles instead of the Gyms.

### CODE REVIEW

**Comment by Someonealive-QN** [comment]
Like this? 
86 Sem Título_20220514112541
 Ins't mine but the person who made them said its free to use 
Screenshot_20220514-111344
Here

**Comment by Ultima1990** [comment]
> Like this? 86 Sem Título_20220514112541 Ins't mine but the person who made them said its free to use Screenshot_20220514-111344 Here

Link?

**Comment by Someonealive-QN** [comment]
Here:
https://www.pokecommunity.com/showthread.php?p=9849231

File: src/scripts/Update.ts [hunk 1]
```diff
@@ -716,6 +716,12 @@ class Update implements Saveable {
             // Remove the Elite_Nihilego Gym, now a temporary battle instead of a gym
             saveData.statistics.gymsDefeated.splice(84, 1);
             saveData.badgeCase.splice(84, 1);
+            // Change Ultra Megalopolis to a Temporary Battle
+            saveData.statistics.temporaryBattleDefeated = new Array<number>();
```
Comment (Jaaslet) [inline src/scripts/Update.ts:720]: This line needs to be removed. It's overwriting the Nihilego temp battle update

File: src/scripts/Update.ts [hunk 2]
```diff
@@ -716,6 +716,12 @@ class Update implements Saveable {
...
+            saveData.statistics.temporaryBattleDefeated[1] = saveData.statistics.gymsDefeated[89];
```
Comment (Jaaslet) [inline src/scripts/Update.ts:721]: Because Nihilego have been removed earlier in the update function, this gym now has index 88 and not index 89

**Review by Jaaslet — CHANGES_REQUESTED** [review]
You also need to remove 'Elite_UltraNecrozma' from badges.ts for it to work.
I have reviewed the code, tested the update function and tested it as a lategame player.
I have not yet tested if the quest still works

**Review by Jaaslet — APPROVED** [review]
Looks good.
The update function now works and i have now also played though the quest as a new Alola player. It works just as before.
So i have no more comments

**Pull request merged**